### ISSUE TITLE
Set holding time/duration for wave file

### ISSUE THREAD

**Issue by jerrojelly** [alemangui/pizzicato#48]
We can modify attack and release time, how about modifying duration? I do often have unequal wave file duration e.g. preview audio track that will play 5 seconds (offset fade-in, duration: 5 secs and release fade-out) instead of full length.

**Comment by jerrojelly** [comment]
Any comment on the holding time?

**Comment by alemangui** [comment]
Sorry for the delay, hectic week. Can you elaborate on the use case? 

Are we talking about specifying which segment of an audio track to use in a ```Pz.Sound```? E.g., I have a 20 second mp3 file, of which I want to use only the first 8 seconds and ignore the rest? If so, this may be a starting point for a discussion on audio sprites.

**Comment by jerrojelly** [comment]
Yes that right, I have some mp3 audio files that will only need to play for a few seconds and fade out in the last few seconds.

How does chopped up sound look like?

**Comment by jerrojelly** [comment]
Was there a parameter to set the fadeout durstion in 1 or 2 seconds to as long as 10 seconds?

**Comment by alemangui** [comment]
Ok, it could be interesting to get this feature into the library. I'm not sure how the API should look like though.

## Through the Sound constructor

On one hand we could add parameters in the constructor of a ```file``` type sound to specify the time range it should act on. An example of this would be ```start``` and ```end```, or ```range```. It could look like this:

```javascript
const sound = new Pz.Sound({ 
    source: 'file',
    options: { 
        path: './audio/sound.wav',
        range: [0, 1200]
})
```

The only issue is if we loaded a single audio file containing many sprites, we would have to repeat the above code for each sprite. We could mitigate this by adding a helper function but the root of the problem remains.

## Adding the sprites directly into the Sound object

This would be a bigger change. Basically allowing a Sound object to contain sprites. A very rough draft of an API could be:

```javascript
const sound = new Pz.Sound({ 
    source: 'file',
    options: { 
        path: './audio/sound.wav',
        sprites: {
            drums: [0, 1200],
            guitar: [1200, 2000]
        }
})
```
And then find a way to play each sprite. Perhaps something like 
```javascript
sound.play('drums')
```

This is not ideal because the ```play``` function already receives two parameters: ```when``` and ```offset```. One could argue that the sprite is in fact a way of specifying ```when``` an ```offset```.   I'm at this moment unsure of what would be the best way to play a sprite within a Sound.

What do you think?

**Comment by jerrojelly** [comment]
It looks fine and easy to define range, to make it more useful I have come up easier way to add 

```
const sound = new Pz.Sound({ 
    source: 'file',
    options: { 
        path: './audio/sound.wav',
        range: [0, 1200],
        fadein: [10, 100, "gradual"],
        fadeout: [1000, 200, "gradual"]
})
```

fadein start at 10ms and reach from 0-100% volume in 100ms
fadein start at 1000ms and reach from 100-0% volume in 200ms

If there is a user defined volume we could set during playback,
Set to 90% volume at 500ms (e.g.  100% to 90% within 100ms), 50% volume at 800ms  (e.g.  90% to 50% within 100ms):
sound.volumeplayback = {
   "500": [90, 100, "gradual"]
   "800": [50, 100, "instant"]
}

**Comment by alemangui** [comment]
In your previous example:

```javascript
const sound = new Pz.Sound({ 
    source: 'file',
    options: { 
        path: './audio/sound.wav',
        range: [0, 1200],
        fadein: [10, 100, "gradual"],
        fadeout: [1000, 200, "gradual"]
})
```

What would happen when the sound is playing from 0 to 10ms? Would it be silent?

If so, I am not convinced about having to specify fade-in's start time. The reason is that if the initial segment of a sound should be silent, the parameters ```when``` and ```offset``` of the ```play()``` function should be used. 

For instance, in the above case, to ensure that the sound starts playing at 10ms you could do:

```javascript
sound.play(0, 0.01)
```

That way the attack/release parameters remain simple.

**Comment by jerrojelly** [comment]
0-10 ms is not silent.

**Comment by alemangui** [comment]
OK in that case the sprites would suffice. I want to make sure the API is not complexified by adding things in rather edge cases that can also be achieved differently. 

So let's use this Github issue for the sprites feature.

**Comment by jerrojelly** [comment]
That's great!
